### PR TITLE
LVIIICNET-142 Feature/projects details 

### PR DESCRIPTION
**Pull request by artur-britvin**

### CODE REVIEW

File: TheraLang.BLL/Services/ProjectParticipationService.cs [hunk 1]
```diff
@@ -103,5 +126,7 @@ public async Task CreateRequest(Guid userId, int projectId)
                                     $"and {nameof(projectId)}:{projectId}: ", ex);
             }
         }
+
```
Comment (Dreylak) [inline TheraLang.BLL/Services/ProjectParticipationService.cs]: Say no to empty lines

File: TheraLang.Web/ClientApp/src/app/modules/main/pages/project/project-info/project-info.component.html [hunk 2]
```diff
@@ -1,25 +1,94 @@
...
 
-    <img mat-card-image class="ProjImg" src="../../assets/img/hatiko.gif" alt="Photo">
-    <mat-card-content>
-      <p class="description">{{projectInfo.description}}</p>
-    </mat-card-content>
-  </mat-card>
-  <div class="bottom-buttons">
-    <button mat-raised-button class="resButton" (click)="onJoin()">{{'components.projects.join'|translate}}</button>
-    <button mat-raised-button class="resButton" (click)="getResourcesData()">{{'components.resources.title'|translate}}</button>
+    <div class="actions-conteiner" fxLayout="row">
```
Comment (Dreylak) [inline TheraLang.Web/ClientApp/src/app/modules/main/pages/project/project-info/project-info.component.html]: contAiner

File: TheraLang.Web/ClientApp/src/app/modules/main/pages/project/project-info/project-info.component.ts [hunk 3]
```diff
@@ -1,52 +1,99 @@
...
+        "open",
+        style({
+          display: "initial",
         })
       ),
       state(
         "closed",
         style({
           display: "none"
+
```
Comment (Dreylak) [inline TheraLang.Web/ClientApp/src/app/modules/main/pages/project/project-info/project-info.component.ts]: No empty lines

File: TheraLang.Web/ClientApp/src/app/app-routing.module.ts [hunk 4]
```diff
@@ -33,11 +33,7 @@ const routes: Routes = [
         component: ProjectFormComponent,
         canActivate: [AuthGuard]
       },
-      {
-        path: "projects/:id",
-        component: ProjectInfoComponent,
-        canActivate: [AuthGuard]
-      },
+      { path: "projects/:id", component: ProjectInfoComponent},
```
Comment (BohdanZahirnuy) [inline TheraLang.Web/ClientApp/src/app/app-routing.module.ts]: Why did you delete CanActivate fucntion ?

File: TheraLang.BLL/Services/ProjectParticipationService.cs [hunk 5]
```diff
@@ -69,6 +70,28 @@ public async Task<IEnumerable<ProjectParticipationDto>> GetAll()
...
+        {
+            var projectParticipations = await _unitOfWork.Repository<ProjectParticipation>()
+                .GetAll()
+                .Where(p=>p.ProjectId == projectId && p.Status == ProjectParticipationStatus.Approved)
+                .Include(p => p.User)
+                .Include(p => p.Project)
+                .ToListAsync();
+
+            var mapper = new MapperConfiguration(cfg => cfg.CreateMap<ProjectParticipation, ProjectParticipationDto>()
+                .ForMember(m => m.ProjectId, opt => opt.MapFrom(m => m.ProjectId))
```
Comment (mod1fy) [inline TheraLang.BLL/Services/ProjectParticipationService.cs]: No need to specify if the name of fields the same

File: TheraLang.Web/ClientApp/src/app/app-routing.module.ts (hunk 4 above)
Comment (artur-britvin) [reply TheraLang.Web/ClientApp/src/app/app-routing.module.ts]: Cuz guest will can visit to details page

File: TheraLang.BLL/Services/ProjectParticipationService.cs (hunk 1 above)
Comment (artur-britvin) [reply TheraLang.BLL/Services/ProjectParticipationService.cs]: Done

File: TheraLang.Web/ClientApp/src/app/modules/main/pages/project/project-info/project-info.component.html (hunk 2 above)
Comment (artur-britvin) [reply TheraLang.Web/ClientApp/src/app/modules/main/pages/project/project-info/project-info.component.html]: done

File: TheraLang.Web/ClientApp/src/app/modules/main/pages/project/project-info/project-info.component.ts (hunk 3 above)
Comment (artur-britvin) [reply TheraLang.Web/ClientApp/src/app/modules/main/pages/project/project-info/project-info.component.ts]: done

File: TheraLang.BLL/Services/ProjectParticipationService.cs (hunk 5 above)
Comment (artur-britvin) [reply TheraLang.BLL/Services/ProjectParticipationService.cs]: done

File: TheraLang.BLL/Services/ResourceService.cs [hunk 6]
```diff
@@ -189,28 +190,18 @@ public async Task<IEnumerable<ResourceDto>> GetAllResourcesByProjectId(int proje
...
-                        ResourceProjects = res.ResourceProjects,
-                        UpdatedById = res.UpdatedById,
-                        CreatedDateUtc = res.CreatedDateUtc,
-                        UpdatedDateUtc = res.UpdatedDateUtc,
-                    }).ToList();
+                    .GetAll()
+                    .Where(x => x.ResourceProjects.Any(c => c.ProjectId == projectId))
+                    .Include(x => x.ResourceCategory)
+                    .ToListAsync();
+
```
Comment (Dreylak) [inline TheraLang.BLL/Services/ResourceService.cs]: Say no to this empty lines

File: TheraLang.Web/ViewModels/ResourceViewModel.cs [hunk 7]
```diff
@@ -13,9 +14,11 @@ public class ResourceViewModel
         public string Url { get; set; }
 
         public string FileName { get; set; }
-        
+
         public IFormFile File { get; set; }
 
         public int CategoryId { get; set; }
+        public DateTime CreatedDateUtc { get; set; }
```
Comment (Dreylak) [inline TheraLang.Web/ViewModels/ResourceViewModel.cs]: Add empty line here or remove them above
